package user_search;
message PbUserSearchQuery{
    message TermExpansion{
        required bytes term = 1;
        repeated bytes eterm = 2;
    }
    message LinkDistanceFilter{
        optional bool link_1du = 1;
        optional bool link_2du = 2;
        optional bool link_xdu = 3;
    }
    
    optional int32 offset = 1;
    optional int32 count = 2;
   
    required bytes xapian_query_blob = 3;

    optional int64 searcher_uid = 5;
    repeated int64 d1_uid = 6;
    repeated int64 d2_uid = 7;
    
    repeated TermExpansion te = 8;
    
    optional int32 search_type = 9; //normal_search 1, jd_matching 2, user_matching 3
    
    optional LinkDistanceFilter link_distance_filter = 10;; // bit sets

    repeated uint64 stats_com_term = 13;  //reserved terms for stats
    repeated uint64 stats_job_term = 14;
    repeated uint64 stats_tag_term = 15;
    repeated uint64 stats_sch_term = 16;

    optional int32 task_type = 17;     
    
    optional bytes searcher_company_name = 19;
    optional bool filter_company_1du_contacts = 20;
    optional bool filter_non_company_1du_contacts = 21;

    optional bool debug = 18;
    
}



